### PR TITLE
Implement TizenArtifacts and unsupport AZURE_BUILD_ID

### PR DESCRIPTION
**Pull request by swift-kim**
Part of https://github.com/flutter-tizen/flutter-tizen/issues/286 and https://github.com/flutter-tizen/engine/issues/224.

Reland `TizenArtifacts` which has previously been removed by https://github.com/flutter-tizen/flutter-tizen/pull/150, to enable overriding  `gen_snapshot` paths for Tizen during app build.

Notes:
- Only Linux `gen_snapshot` artifacts are currently included in the [Tizen engine 2.8.0 (1) pre-release](https://github.com/flutter-tizen/engine/releases/tag/790d2b0). I'll update the artifacts when https://github.com/flutter-tizen/engine/issues/224 is implemented by @WonyoungChoi. If you are on Windows or macOS, you can manually build `gen_snapshot` and copy it to the cached artifact directory corresponding to your host platform (e.g. `flutter/bin/cache/artifacts/engine/tizen-arm-release/windows-x64`).
- If anyone is using **flutter-tizen** to build **Android** apps, beware that this change may break Android release and profile builds. This is inevitable because we use the same `TargetPlatform` names for Android and Tizen.

### CODE REVIEW

File: lib/tizen_artifacts.dart [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      assert(mode != null, 'Need to specify a build mode.');
+      assert(mode != BuildMode.debug,
+          'Artifact $artifact only available in non-debug mode.');
+      final String arch =
+          _getArchForTargetPlatform(platform ?? TargetPlatform.android_arm);
+      final String hostPlatform =
+          getNameForHostPlatform(getCurrentHostPlatform());
+      return _getEngineArtifactsDirectory(arch, mode!)
+          .childDirectory(hostPlatform)
+          .childFile('gen_snapshot')
```
Comment (WonyoungChoi) [inline lib/tizen_artifacts.dart:53]: Do we need ".exe" suffix on windows?

File: lib/tizen_artifacts.dart [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    EnvironmentType? environmentType,
+  }) {
+    if (artifact == Artifact.genSnapshot) {
+      assert(mode != null, 'Need to specify a build mode.');
+      assert(mode != BuildMode.debug,
+          'Artifact $artifact only available in non-debug mode.');
+      final String arch =
+          _getArchForTargetPlatform(platform ?? TargetPlatform.android_arm);
+      final String hostPlatform =
+          getNameForHostPlatform(getCurrentHostPlatform());
```
Comment (WonyoungChoi) [inline lib/tizen_artifacts.dart:50]: We need to check the `supported host platform` here. `linux-arm64`, `darwin-arm` are out of our attention now. `darwin-arm` may be need to support `Apple Silicon` later.

Comment (WonyoungChoi) [inline lib/tizen_artifacts.dart:50]: We need to check the `supported host platform` here. we don't support `linux-arm64` now.

**Comment by WonyoungChoi** [comment]
Now that `tizen-arm-release.zip` contains the `gen_snapshot` binary, but when the CI works are finished, the artifacts will be separated to download only the artifact required for the host platform.

File: lib/tizen_artifacts.dart (hunk 1 above)
Comment (swift-kim) [reply lib/tizen_artifacts.dart:53]: I believe we don't. I'll double check later when gen_snapshots for Windows become available.

File: lib/tizen_artifacts.dart (hunk 2 above)
Comment (swift-kim) [reply lib/tizen_artifacts.dart]: I added an assert although it's not very meaningful because asserts are not enabled for `dart run` by default.

**Pull request merged**